pkg/command/client.go

```diff
@@ -65,38 +70,51 @@ func (c *Client) SyncDashboard(ctx context.Context, uid string, queriesDir strin
 		if panel.Description == nil {
 			continue
 		}
-		queryName := ""
-		for _, part := range strings.Split(*panel.Description, "\n") {
-			queryParts := strings.Split(part, "query=")
-			if len(queryParts) != 2 {
-				continue
-			}
-			queryName = queryParts[1]
-		}
-		if queryName == "" {
+
+		targets := panel.GetTargets()
+		if targets == nil || (targets != nil && len(*targets) == 0) {
+			log.Printf("[%s:%s] panel has no targets found", panel.Type, panel.Title)
 			continue
 		}
 
-		var queryPath string
-		// support query name with and without .sql extension
-		if queryPath, _ = queries[queryName]; queryPath == "" {
-			if queryPath, _ = queries[fmt.Sprintf("%s.sql", queryName)]; queryPath == "" {
+		for i, part := range strings.Split(*panel.Description, "\n") {
+			queryName := ""
+			queryParts := strings.Split(part, "query=")
+			if len(queryParts) == 2 {
+				queryName = queryParts[1]
+			}
+
+			t := *panel.GetTargets()
+			q := queries.Get(queryName)
+			if q == nil {
 				log.Printf("[%s:%s] query %s not found", panel.Type, panel.Title, queryName)
 				continue
 			}
-		}
-		queryBy, err := ioutil.ReadFile(queryPath)
-		if err != nil {
-			return err
-		}
 
-		// TODO(dm): support multiple targets?
-		targets := panel.GetTargets()
-		if targets != nil && len(*targets) > 0 {
-			t := *targets
-			t[0].RawSql = string(queryBy)
+			if i < len(t) {
+				switch q.Type {
+				case SQL:
+					t[i].RawSql = q.Raw
+					log.Printf("target updated: [%s:%s] query[%d] %s", panel.Type, panel.Title, i, queryName)
+				case Prometheus:
+					t[i].Expr = q.Raw
+					log.Printf("target updated: [%s:%s] query[%d] %s", panel.Type, panel.Title, i, queryName)
+				}
+			} else {
```

Comment:
We might need to split the query parsing and target setting into two steps.
If the description has more lines, the comparison of the **targets** len and **i** will fail.

At a high level, we could apply the following:
* Parse all the queries from the description
* Map the queries; _in order_; to the existing targets
* Create new targets for the queries that don't yet have a target